docs/architecture/matching-service.md

```diff
@@ -5,9 +5,15 @@
 <!-- https://lucid.app/lucidchart/0202e4b8-5258-4cd6-a6a0-67159300532b/edit -->
 <img src="../_assets/matching-context.svg">
 
+## Task Queues
 Matching Service instances manage [Task Queues](https://docs.temporal.io/workers#task-queue) being polled by Temporal Worker processes.
 Long-poll requests from Temporal Workers are received by the Frontend Service, which routes them to the Matching Service instance responsible for the requested Task Queue.
 The Matching Service instance responds by sending Workflow Tasks and Activity Tasks from the requested Task Queue.
 A single Task Queue is responsible for delivering tasks relating to many Workflow Executions.
 
-Documentation of Matching Service internals is not yet available.
+### Task Queue Partitions
+Matching Service splits Task Queues into partitions to provide higher throughput overall. Default is 4, but there can be more. Partitions can be redistributed, loaded from storage, and unloaded to storage. Unloading/loading will preserve any existing backlog of tasks in a TaskQueue.
```

Comment:
It doesn't sound correct to say "partitions can be loaded from storage and unloaded from storage". The partition is more like the process responsible for matching tasks, it's not a durable entity itself. You could say that about partition _metadata_, or the partition backlog (though I wouldn't say that is "unloaded to storage" either, it's created and sits there).

Reply:
Ahhh, thank you. I didn't feel quite certain about how I said this. Changed.